src/python/zquantum/core/measurement.py

```diff
@@ -446,6 +447,36 @@ def from_counts(cls, counts: Dict[str, int]):
         measurements.add_counts(counts)
         return measurements
 
+    @classmethod
+    def get_measurements_representing_distribution(
+        cls, bitstring_distribution, number_of_samples
+    ):
+        distribution = copy.deepcopy(bitstring_distribution.distribution_dict)
+
+        bitstring_samples = []
+        for state in distribution:
+            bitstring = tuple([int(measurement_value) for measurement_value in state])
+
+            bitstring_samples += [bitstring] * int(
+                distribution[state] * number_of_samples
+            )
+
+        while len(bitstring_samples) != number_of_samples:
```

Comment:
Can we delete the while loop since the needed bitstrings will get added during the first iteration?

Reply:
Addressed here: 2fb7d11